schemas/manifest.schema.json

```
@@ -0,0 +1,473 @@
+{
+    "$schema": "http://json-schema.org/draft-07/schema",
+    "$id": "https://github.com/YunoHost/apps/blob/master/schemas/manifest.schema.json",
+    "title": "Yunohost app manifest.toml schema",
+    "version": "0",
+
+    "type": "object",
+    "$defs": {
+        "translated_string": {
+            "type": "object",
+            "required": ["en"],
+            "additionalProperties": false,
+            "patternProperties": {
+                "^[a-z]{2}$": {
+                    "type": "string"
+                }
+            }
+        },
+        "byte_size": {
+            "type": "string",
+            "pattern": "^[0-9]*(\\.[0-9]*)?[kKmMgG]$"
+        },
+        "boolean_or_not_relevant": {
+            "anyOf": [
+                {
+                    "type": "boolean"
+                },
+                {
+                    "type": "string",
+                    "const": "not_relevant"
+                }
+            ]
+        },
+        "string_or_array_of": {
+            "anyOf": [
+                {
+                    "type": "string"
+                },
+                {
+                    "type": "array",
+                    "items": {
+                        "type": "string"
+                    }
+                }
+            ]
+        }
+    },
+
+    "properties": {
+        "packaging_format": {
+            "description": "Packaging format / version",
+            "type": "integer"
+        },
+        "id": {
+            "description": "App ID",
+            "type": "string"
+        },
+        "name": {
+            "description": "App Name",
+            "type": "string"
+        },
+        "version": {
+            "description": "App and package version",
+            "type": "string",
+            "pattern": "^[0-9\\.]*~ynh[0-9]*"
+        },
+        "description": {"$ref": "#/$defs/translated_string"},
+        "maintainers": {
+            "description": "List of maintainers",
+            "type": "array",
+            "items": {
+              "type": "string"
+            }
+        },
+        "upstream": {
+            "type": "object",
+            "required": ["license"],
+            "additionalProperties": false,
+            "properties": {
+                "license": {"type": "string"},
+                "website": {"type": "string", "format": "uri-reference"},
+                "demo": {"type": "string", "format": "uri-reference"},
+                "admindoc": {"type": "string", "format": "uri-reference"},
+                "userdoc": {"type": "string", "format": "uri-reference"},
+                "code": {"type": "string", "format": "uri-reference"},
+                "fund": {"type": "string", "format": "uri-reference"},
+                "cpe": {
+                    "type": "string",
+                    "pattern": "^cpe:[0-9\\.]*:[a-z]*:[a-zA-Z0-9_]*:[a-z]*$"
```

Hmmm so there's a couple minor issues with this ... I think we don't have the "end part" of the regex which if i recall correctly contains the version number or something ... Also we should allow for `-` in the name

```
In key : upstream > cpe
'cpe:2.3:a:home-assistant:home-assistant' does not match '^cpe:[0-9\\.]*:[a-z]*:[a-zA-Z0-9_]*:[a-z]*$'
In key : upstream > cpe
'???' does not match '^cpe:[0-9\\.]*:[a-z]*:[a-zA-Z0-9_]*:[a-z]*$'
In key : upstream > cpe
'cpe:2.3:a:simplex:simplex_chat' does not match '^cpe:[0-9\\.]*:[a-z]*:[a-zA-Z0-9_]*:[a-z]*$'
In key : upstream > cpe
'cpe:2.3:a:pfsense:pfsense-pkg-wireguard' does not match '^cpe:[0-9\\.]*:[a-z]*:[a-zA-Z0-9_]*:[a-z]*$'
```

I think I would go for a more constraining one for now (anyway this field ain't really used for now) : 

```suggestion
                    "pattern": "^cpe:2.3:a:[a-zA-Z0-9_-]*:[a-zA-Z0-9_-]*$"
```